src/execution/collectFields.ts

```diff
@@ -129,21 +133,24 @@ function collectFieldsImpl(
   const {
     schema,
     fragments,
-    variableValues,
     runtimeType,
+    variableValues,
+    localVariableValues,
     operation,
     visitedFragmentNames,
   } = context;
 
   for (const selection of selectionSet.selections) {
     switch (selection.kind) {
       case Kind.FIELD: {
-        if (!shouldIncludeNode(variableValues, selection)) {
+        const vars = localVariableValues ?? variableValues;
```

Comment:
why not both? as `getDirectiveValues` ultimately calls `getArgumentValues`, might it not be necessary to track both the global and local variableValues through? This should be true whenever/wherever directives are used.

Reply:
Hmm, I assume this comment is misplaced then, you mean that https://github.com/JoviDeCroock/graphql-js/pull/2/files#diff-1d705c6a4c73cd3ce46190029e75abd3015b49de3ec250357dcf9b44a35cb7d0R931 should become `{ ...exeContext.variableValues, ...fragmentVariableValues }`?

Reply:
Maybe that should be looked at as well... But the problem I am referencing here --- somewhat circuitously --- is that `shouldIncludeNode` passes the variableValues to `getDirectiveValues` which then calls `getArgumentValues`. `getArgumentValues` requires both the local and the global variableValues, but it isn't receiving them.

Overall, I wonder if we could consider abandoning the coalescence of globalVariableValues into the localVariables; instead I think it would be better to simply retain references to both the global and local variables, and have the getArgumentValues function perform whatever analysis it needs. I believe also the local variable definitions/signatures would also have to be passed to getArgumentVariables and I would suggest a single additional argument like:

```ts
interface ExperimentalFragmentVariables {
   signatures: ObjMap<GraphQLFragmentVariableSignatures>;   // the defs with the types validated and the defaults calculated
   variableValues: ObjMap<unknown>;
} 
```

The idea of preserving the signatures/definitions in an interface is similar to the approach by @leebyron with respect to default value validation (see mega-PR https://github.com/graphql/graphql-js/pull/3049  and specifically https://github.com/graphql/graphql-js/pull/3077/files#diff-777202fe667b9e656f40631c341c7c0776895a75eff7004578bf9b7a97e095abR29-R37) and may be the direction we will be heading in anyway whenever that long awaited work is merged.

Reply:
[This PR](https://github.com/JoviDeCroock/graphql-js/pull/9) demonstrates what I believe is a test failure. Let me know what you think.

Your first comment could have a potential fix to that, but the link is not working for me right now, I think it pointed to here: https://github.com/JoviDeCroock/graphql-js/pull/2/files#diff-1d705c6a4c73cd3ce46190029e75abd3015b49de3ec250357dcf9b44a35cb7d0L726, which would be only for the stream directive, although you could do the same thing for skip/include, but I think that would also not work, as to deal with the shadowing case, `getArgumentValues` has to know that there was a fragmentVariable defined and not provided, so `getDirectiveValues` would have to know that there are actually fragmentVariables just like the call to `getArgumentValues` passes fragmentVariables in.